compose.py

```diff
@@ -0,0 +1,276 @@
+        'Generated dockerfiles are placed here. Default to current directory.'
+    )
+    parser.add_argument(
+        '--version',
+        type=str,
+        required=False,
+        help=
+        'The Triton version. If not specified defaults to the value in the TRITON_VERSION file.'
+    )
+    parser.add_argument(
+        '--container-version',
+        type=str,
+        required=False,
+        help=
+        'The version to use for the generated Docker image. If not specified the container version will be chosen automatically based on --version value.'
+    )
+    parser.add_argument(
+        '--upstream-container-version',
+        type=str,
+        required=False,
+        help=
+        'The version of the NGC Docker images to use as a base for this image. If not specified the upstream container version will be chosen automatically based on --version value.'
+    )
+    parser.add_argument(
+        '--enable-gpu',
+        action="store_true",
+        required=False,
+        help='Generate a Triton image that supports GPU'
+    )
+    parser.add_argument(
+        '--backend',
+        action='append',
+        required=False,
+        help=
+        'Include <backend-name> in the generated Docker image. The flag may be specified multiple times.'
+    )
+    parser.add_argument(
+        '--repoagent',
+        action='append',
+        required=False,
+        help='Include <repoagent-name> in the generated Docker image. The flag may be specified multiple times.')
+    FLAGS = parser.parse_args()
+    
+    if FLAGS.work_dir is None:
+        FLAGS.work_dir = "."
+    if FLAGS.output_name is None:
+        FLAGS.output_name = "tritonserver"
+    # Determine the versions. Start with Triton version, if --version
+    # is not explicitly specified read from TRITON_VERSION file.
+    if FLAGS.version is None:
+        with open('TRITON_VERSION', "r") as vfile:
+            FLAGS.version = vfile.readline().strip()
+    if FLAGS.container_version is None:
+        if FLAGS.version not in TRITON_VERSION_MAP:
+            fail('container version not known for {}'.format(FLAGS.version))
+        FLAGS.container_version = TRITON_VERSION_MAP[FLAGS.version][0]
+    if FLAGS.upstream_container_version is None:
+        if FLAGS.version not in TRITON_VERSION_MAP:
+            fail('upstream container version not known for {}'.format(
+                FLAGS.version))
+        FLAGS.upstream_container_version = TRITON_VERSION_MAP[FLAGS.version][1]
+
+    dockerfile_name = 'Dockerfile.compose'
+
+    if FLAGS.backend is None:
+        FLAGS.backend = []
+    if FLAGS.repoagent is None:
+        FLAGS.repoagent = []
+
+    start_gpu_dockerfile(FLAGS.work_dir, dockerfile_name)
+    add_requested_backends(FLAGS.work_dir, dockerfile_name, FLAGS.backend)
+    add_requested_repoagents(FLAGS.work_dir, dockerfile_name, FLAGS.repoagent)
+
+    upstreamDockerImage = 'nvcr.io/nvidia/tritonserver:{}-py3'.format(FLAGS.upstream_container_version)
```

Comment:
Put this in a function (like create_argmap()). You will need to call it before start_gpu_dockerfile() because you need the version information in that function.